fmriprep/workflows/confounds.py

```diff
@@ -243,3 +264,211 @@ def reverse_order(inlist):
     if isinstance(inlist, list):
         inlist.reverse()
     return inlist
+
+
+def get_ica_confounds(ica_out_dir, ignore_aroma_err):
+    import os
+    import shutil
+    import numpy as np
+    from niworkflows.nipype import logging
+
+    # To catch edge cases when there are either no noise or signal components
+    LOGGER = logging.getLogger('workflow')
+
+    # Pass in numpy array and column base name to generate headers
+    # modified from add_header_func
+    def aroma_add_header_func(np_arr, col_base, comp_nums):
+        import pandas as pd
+        from sys import version_info
+        PY3 = version_info[0] > 2
+
+        df = pd.DataFrame(np_arr, columns=[str(col_base) + str(index) for index in comp_nums])
+        df.to_csv(str(col_base) + "AROMAConfounds.tsv",
+                  sep="\t" if PY3 else '\t'.encode(), index=None)
+
+        return os.path.abspath(str(col_base) + "AROMAConfounds.tsv")
+
+    # load the txt files from ICA_AROMA
+    melodic_mix = os.path.join(ica_out_dir, 'melodic.ica/melodic_mix')
+    motion_ics = os.path.join(ica_out_dir, 'classified_motion_ICs.txt')
+
+    # Change names of motion_ics and melodic_mix for output
+    melodic_mix_out = os.path.join(ica_out_dir, 'MELODICmix.tsv')
+    motion_ics_out = os.path.join(ica_out_dir, 'AROMAnoiseICs.csv')
+
+    # melodic_mix replace spaces with tabs
+    with open(melodic_mix, 'r') as melodic_file:
+        melodic_mix_out_char = melodic_file.read().replace('  ', '\t')
+    # write to output file
+    with open(melodic_mix_out, 'w+') as melodic_file_out:
+        melodic_file_out.write(melodic_mix_out_char)
+
+    # copy metion_ics file to derivatives name
+    shutil.copyfile(motion_ics, motion_ics_out)
+
+    # -1 since python lists start at index 0
+    motion_ic_indices = np.loadtxt(motion_ics, dtype=int, delimiter=',') - 1
+    melodic_mix_arr = np.loadtxt(melodic_mix, ndmin=2)
+
+    # Return dummy list of ones if no noise compnents were found
+    if motion_ic_indices.size == 0:
+        if ignore_aroma_err:
+            LOGGER.warn('WARNING: No noise components were classified')
+            no_noise_arr = np.ones((melodic_mix_arr.shape[0], 1))
+            aggr_tsv = aroma_add_header_func(no_noise_arr, 'AROMANoNoiseAggr', ['00'])
+            aroma_confounds = aggr_tsv
+            return aroma_confounds, motion_ics_out, melodic_mix_out
+        else:
+            raise RuntimeError('ERROR: ICA-AROMA found no noise components!')
```

Comment:
I vaguely remember there being some discussion earlier, but it's hard to find at this point.

Why are we returning a dummy vector as opposed to one with `size == 0`, or `None`?